File: schemas/project-management/convex/time_entries.ts
// timeEntries: logged time spent on tasks by users.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const timeEntries = defineTable({
  taskId: v.id("tasks"),
  userId: v.id("users"),
  description: v.optional(v.string()),
  startTime: v.optional(v.number()),
  endTime: v.optional(v.number()),
  duration: v.number(),
  updatedAt: v.number(),
})
  .index("by_task_id", ["taskId"])
  .index("by_user_id_start_time", ["userId", "startTime"]);

File: schemas/project-management/spacetimedb/time_entries.rs
// time_entries: Logged time spent by users working on tasks.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = time_entries, public)]
pub struct TimeEntry {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub task_id: String, // UUID — FK → tasks.id (cascade delete)
    #[index(btree)]
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    pub description: Option<String>,
    pub start_time: Option<Timestamp>,
    pub end_time: Option<Timestamp>,
    pub duration: i32,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
    // Composite index: (user_id, start_time)
}
